src/fits.cc

```diff
@@ -825,6 +825,9 @@ void MetadataIterationFunctor::operator()(std::string const &key, std::string co
     static boost::regex const intRegex("[+-]?[0-9]+");
     static boost::regex const doubleRegex("[+-]?([0-9]*\\.?[0-9]+|[0-9]+\\.?[0-9]*)([eE][+-]?[0-9]+)?");
     static boost::regex const fitsStringRegex("'(.*?) *'");
+    // regex for two-line comment added to all FITS headers by CFITSIO
+    static boost::regex const fitsDefinitionCommentRegex(
```

Comment:
I was thinking about std::regex as well, but then saw the previous lines so decided not to comment. It would be nice if we could get rid of boost::regex in afw since there is not much of it (and when pex_policy goes, most of boost::regex usage will go).

Reply:
FYI there are 3 files in afw that use boost::regex and 2 of those are involved in FITS I/O (the other is one usage in warpExposure.cc.

Reply:
I'm afraid I had to back out the switch to `std::regex` in the end - it caused a test failure in Jenkins on OS X only.  It looks like the behavior of libc++'s std::regex is not the same as boost's, while libstdc++'s is.  I'm not sure which one is the same as the C++ standard, but I'm not in the mood to go debugging regular expression syntax right now.